### DISCUSSION TITLE
Android: Running detekt with type resolution on whole code base

### DISCUSSION THREAD

**Discussion by ppospisil-paylocity** [detekt/detekt#7538] [Q&A] [answered]
Do I understand correctly I need to run 3 Gradle tasks to have covered the whole project (production code, unit tests, android tests) by detekt with type resolution?

For build variant AB it would be **detektAP**, **detektABUnitTest**, **detektABAndroidTest**.
As far as I understand the configuration without the compiler plugin, it would mean 3 builds, right?

My post is based on
https://youtu.be/6EQd_SDR6n0?si=rXcI0V31Ybm9xiVg&t=906 *Runs another compilation Internally to resolve types*
and https://detekt.dev/docs/gettingstarted/gradle

My expectation is to run one Gradle task to check the whole project (with type resolution).

**Comment by BraisGabin** [comment, marked as answer]
The only way to avoid the double compilation (it's not truly a compilation but the time consumed is similar) is to use the compiler plugin. You can't execute all at the same time because the classpath for each of those task are different. If you want type solving (and that's good) you need correct classpaths and for that you should execute those 3 tasks.

As an extra I recommend you to create a "meta task" that depends on the task that you care of and call it `detektAll` or something similar. It's handy. That's what o have in my android projects. On jvm projects it's way easier because it's just `detektMain` and `detektTest`.

**Reply by ppospisil-paylocity** [reply]
Thank you very much ❤️ Marking as answer ✅
